lua/pixelui/core/cl_color.lua

```diff
@@ -45,96 +28,85 @@ function PIXEL.Hue2RGB(p, q, t)
     return p
 end
 
-function PIXEL.HSLToColor(h, s, l, a)
-    local r, g, b
-    local t = h / (2 * math.pi)
-
-    if s == 0 then
-        r, g, b = l, l, l
-    else
-        local q
-        if l < 0.5 then
-        q = l * (1 + s)
-        else
-        q = l + s - l * s
-        end
-        local p = 2 * l - q
-
-        r = PIXEL.Hue2RGB(p, q, t + 1 / 3)
-        g = PIXEL.Hue2RGB(p, q, t)
-        b = PIXEL.Hue2RGB(p, q, t - 1 / 3)
-    end
+local createColor = Color
+function PIXEL.CopyColor(col)
+    return createColor(col.r, col.g, col.b, col.a)
+end
 
-    return Color(r * 255, g * 255, b * 255, (a or 1) * 255)
+function PIXEL.OffsetColor(col, offset)
+    return createColor(col.r + offset, col.g + offset, col.b + offset)
 end
 
-function PIXEL.ColorToHSL(col)
-    local r = col.r / 255
-    local g = col.g / 255
-    local b = col.b / 255
-    local max, min = math.max(r, g, b), math.min(r, g, b)
-    b = max + min
-    local h = b / 2
-    if max == min then return 0, 0, h end
-    local s, l = h, h
-    local d = max - min
-    s = l > .5 and d / (2 - b) or d / b
-    if max == r then h = (g - b) / d + (g < b and 6 or 0)
-    elseif max == g then h = (b - r) / d + 2
-    elseif max == b then h = (r - g) / d + 4
+do
+    local format = string.format
+    function PIXEL.DecToHex(dec, zeros)
+        return format("%0" .. (zeros or 2) .. "x", dec)
     end
-    return h * .16667, s, l
-end
 
-function PIXEL.DecToHex(d, zeros)
-    return string.format("%0" .. (zeros or 2) .. "x", d)
+    local max = math.max
```

Comment:
I'm not sure if this is an actual benefit, as the scope here is already extremely small. Just put it in a do ... end around the function.